google/cloud/alloydb/CMakeLists.txt

```diff
@@ -14,100 +14,11 @@
-google_cloud_cpp_find_proto_include_dir(PROTO_INCLUDE_DIR)
-google_cloud_cpp_load_protolist(
-    proto_list
-    "${PROJECT_SOURCE_DIR}/external/googleapis/protolists/alloydb.list")
-google_cloud_cpp_load_protodeps(
-    proto_deps
-    "${PROJECT_SOURCE_DIR}/external/googleapis/protodeps/alloydb.deps")
-google_cloud_cpp_grpcpp_library(
-    google_cloud_cpp_alloydb_protos # cmake-format: sort
-    ${proto_list} PROTO_PATH_DIRECTORIES "${EXTERNAL_GOOGLEAPIS_SOURCE}"
-    "${PROTO_INCLUDE_DIR}")
-external_googleapis_set_version_and_alias(alloydb_protos)
-target_link_libraries(google_cloud_cpp_alloydb_protos PUBLIC ${proto_deps})
-
-file(
-    GLOB source_files
-    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-    ${source_globs})
-list(SORT source_files)
-add_library(google_cloud_cpp_alloydb ${source_files})
-target_include_directories(
-    google_cloud_cpp_alloydb
-    PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-           $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-           $<INSTALL_INTERFACE:include>)
-target_link_libraries(
-    google_cloud_cpp_alloydb
-    PUBLIC google-cloud-cpp::grpc_utils google-cloud-cpp::common
-           google-cloud-cpp::alloydb_protos)
-google_cloud_cpp_add_common_options(google_cloud_cpp_alloydb)
-set_target_properties(
-    google_cloud_cpp_alloydb
-    PROPERTIES EXPORT_NAME google-cloud-cpp::alloydb
-               VERSION "${PROJECT_VERSION}"
-               SOVERSION "${PROJECT_VERSION_MAJOR}")
-target_compile_options(google_cloud_cpp_alloydb
-                       PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
-
-add_library(google-cloud-cpp::alloydb ALIAS google_cloud_cpp_alloydb)
-
-# Create a header-only library for the mocks. We use a CMake `INTERFACE` library
-# for these, a regular library would not work on macOS (where the library needs
-# at least one .o file). Unfortunately INTERFACE libraries are a bit weird in
-# that they need absolute paths for their sources.
-file(
-    GLOB relative_mock_files
-    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-    ${mocks_globs})
-list(SORT relative_mock_files)
-set(mock_files)
-foreach (file IN LISTS relative_mock_files)
-    list(APPEND mock_files "${CMAKE_CURRENT_SOURCE_DIR}/${file}")
-endforeach ()
-add_library(google_cloud_cpp_alloydb_mocks INTERFACE)
-target_sources(google_cloud_cpp_alloydb_mocks INTERFACE ${mock_files})
-target_link_libraries(
-    google_cloud_cpp_alloydb_mocks
-    INTERFACE google-cloud-cpp::alloydb GTest::gmock_main GTest::gmock
-              GTest::gtest)
-set_target_properties(google_cloud_cpp_alloydb_mocks
-                      PROPERTIES EXPORT_NAME google-cloud-cpp::alloydb_mocks)
-target_include_directories(
-    google_cloud_cpp_alloydb_mocks
-    INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-              $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-              $<INSTALL_INTERFACE:include>)
-target_compile_options(google_cloud_cpp_alloydb_mocks
-                       INTERFACE ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
```

Comment:
It may be easier to review all these PRs if you first moved the quickstart section to the bottom of each file.

Reply:
Discussed elsewhere, and the diff is gonna be ugly either way.